### DISCUSSION TITLE
About LLVM version support policy of Enzyme

### DISCUSSION THREAD

**Discussion by PragmaTwice** [EnzymeAD/Enzyme#1212] [General]
It seems currently Enzyme supports a large range of LLVM versions, i.e. from 7 to the latest version (maybe?).

But obviously there are some trade-offs:
- in one hand, supporting more versions will bring enzyme to more users, like users who are using old version of LLVM (but it is sad, I think)
- in the other hand, it will lead to more maintainance cost, e.g. some infra classes or API maybe can only be used in newer versions, but we cannot use them (or at least cannot use them smoothly).

So I wonder is there some LLVM version support policy, like we maintain a dynamic range of LLVM versions we support, and make a timeline of deprecating the old versions and supporting new versions (e.g. from [7, 15] to [8, 16]) to let users know the status of support.

cc @ZuseZ4

**Comment by tgymnich** [comment]
Update: we recently decided to drop support for llvm 7 and 8. Once we have reached feature completeness, we aim to only support the latest llvm version.

**Reply by vchuravy** [reply]
> Once we have reached feature completes, we aim to only support the latest llvm version.

I don't believe we will be in that situation for quite some time. From an Enzyme.jl perspective the lowest version we need support is the latest LTS release (currently Julia v1.6) and I believe LLVM 12

**Reply by ZuseZ4** [reply]
Rust doesn't have LTS, from our perspective we need to support LLVM versions 14-HEAD.
